### DISCUSSION TITLE
Question about the INSBase.c file

### DISCUSSION THREAD

**Discussion by Oops-Qiao** [idaholab/moose#22416] [Q&A General] [answered]
Hello, I am learning the [INSBase.C](https://github.com/idaholab/moose/blob/next/modules/navier_stokes/src/kernels/INSBase.C). But I find the following information. I donot understand why there is a difference when we want to couple velocity and pressure? Why not using   params.addRequiredCoupledVar("p", "pressure"); ?

  // Coupled variables
  params.addRequiredCoupledVar("u", "x-velocity");
  params.addCoupledVar("v", 0, "y-velocity"); // only required in 2D and 3D
  params.addCoupledVar("w", 0, "z-velocity"); // only required in 3D
  **params.addRequiredCoupledVar(NS::pressure, "pressure");
  params.addDeprecatedCoupledVar("p", NS::pressure, "1/1/2022");**

Thanks a lot.

**Comment by GiudGiud** [comment, marked as answer]
Hello

We are requiring pressure with this line, and it's a coupled variable the same way as velocity
```
  params.addRequiredCoupledVar(NS::pressure, "pressure");
```

This other line is for a deprecated old name for the parameter (just p instead of pressure)
params.addDeprecatedCoupledVar("p", NS::pressure, "1/1/2022");

Guillaume
